### PR TITLE
Add answers to multichoice, add method to resolve, a bunch more

### PR DESCRIPTION
**Pull request by LivInTheLookingGlass**

### CODE REVIEW

File: pymanifold/lib.py [hunk 1]
```diff
@@ -149,3 +163,54 @@ def _create_market(
             headers=self._auth_headers(),
         )
         return LiteMarket.from_dict(response.json())
+
+    def resolve_market(self, market: Union[LiteMarket, str], *args, **kwargs):
+        try:
+            market.id
+        except AttributeError:
+            market = self.get_market_by_id(market)
+        match market.outcomeType:
```
Comment (bcongdon) [inline pymanifold/lib.py]: Neat, I haven't used match arms in Python yet. Do you know what the minimum Python version is?

File: pymanifold/lib.py [hunk 2]
```diff
@@ -149,3 +163,54 @@ def _create_market(
             headers=self._auth_headers(),
         )
         return LiteMarket.from_dict(response.json())
+
+    def resolve_market(self, market: Union[LiteMarket, str], *args, **kwargs):
```
Comment (bcongdon) [inline pymanifold/lib.py]: Can you add tests for these new functions?

**Review by bcongdon — COMMENTED** [review]
Thanks for adding this!

File: pymanifold/lib.py (hunk 1 above)
Comment (LivInTheLookingGlass) [reply pymanifold/lib.py]: I actually rewrote that because I was running into versioning problems at the time

**Comment by LivInTheLookingGlass** [comment]
Currently working on tests. Might take me slightly longer to get the ones here, as I'm very used to `pytest`. I assume a lot will translate

Additions to test:
- [ ] get_markets
- [ ] stream_markets
- [ ] list_bets
- [ ] get_bets
- [ ] stream_bets
- [ ] get_market_by_url
- [ ] get_user
- [ ] cancel_market
- [ ] create_multiple_choice_market
- [ ] _resolve_binary_market
- [ ] _resolve_pseudo_numeric_market
- [ ] _resolve_free_response_market
- [ ] _resolve_multiple_choice_market
- [ ] Group
- [ ] LiteUser

**Comment by bcongdon** [comment]
Sounds great! Let me know when you'd like me to take a look

**Comment by LivInTheLookingGlass** [comment]
Okay, I've added tests for maybe half of my additions. Those are certainly ready to check out. I also removed two functions that I had added. They didn't have enough value prop for how hard they were to test. The following is ready for review, but if you'd rather wait until done, I totally get it

- [x] get_markets
- [x] list_groups
- [x] get_groups
- [x] get_group
- [x] list_bets
- [x] get_bets
- [x] get_market_by_url
- [x] get_user
- [x] cancel_market
- [x] create_multiple_choice_market
- [x] _resolve_binary_market
- [x] _resolve_pseudo_numeric_market
- [x] _resolve_free_response_market
- [x] _resolve_multiple_choice_market
- [x] Group
- [x] LiteUser

**Comment by bcongdon** [comment]
So far the implementation LGTM. Looking forward to merging this 👍

**Comment by LivInTheLookingGlass** [comment]
Okay, I think everything I added is good. The only question I have left is if you'd like me to add more of my math utilities to this library instead. I moved the most useful already, but check this file for the rest: https://github.com/LivInTheLookingGlass/ManifoldMarketManager/blob/master/src/util.py

**Comment by LivInTheLookingGlass** [comment]
Everything on  my end is ready. I also have an implementation of the comment API ready to go, but I haven't committed it because the endpoint is returning 401 errors

**Comment by LivInTheLookingGlass** [comment]
Done

File: pymanifold/lib.py [hunk 3]
```diff
@@ -119,35 +229,118 @@ def _create_market(
...
             )
 
         response = requests.post(
             url=BASE_URI + "/market",
             json=data,
             headers=self._auth_headers(),
         )
+        if response.status_code in range(400, 500):
+            response.raise_for_status()
+        elif response.status_code >= 500:
```
Comment (bcongdon) [inline pymanifold/lib.py:271]: Can you add a comment here why if there's an internal error we make a different API call?

**Review by bcongdon — APPROVED** [review]

**Comment by bcongdon** [comment]
LGTM! just had one nitpick comment

**Comment by LivInTheLookingGlass** [comment]
Just a heads up, I accidentally also pushed the comment API. It works and is stable, and has tests, but I wanted to make that a separate PR. Whoops

**Comment by bcongdon** [comment]
Looks like all the tests are passing!
Thanks again for these changes, this is awesome 🎉

**Pull request merged**